src/all/hentaifox/src/eu/kanade/tachiyomi/extension/all/hentaifox/HentaiFox.kt

```diff
@@ -100,4 +113,116 @@ class HentaiFox(
             Filter.Header("HINT: Use double quote (\") for exact match"),
         ) + super.getFilterList().list,
     )
+
+    private val sidebarPath = "includes/sidebar.php"
+
+    private fun sidebarMangaSelector() = "div.item"
+
+    private fun Element.sidebarMangaTitle() =
+        selectFirst("img")?.attr("alt")
+
+    private fun Element.sidebarMangaUrl() =
+        selectFirst("a")?.attr("abs:href")
+
+    private fun Element.sidebarMangaThumbnail() =
+        selectFirst("img")?.imgAttr()
+
+    private fun fetchHomepage(): Request {
+        return GET(baseUrl, headers)
+    }
+
+    private fun homepageCsrfParser(document: Document): String {
+        return document.select("[name=csrf-token]").attr("content")
+    }
+
+    private fun setSidebarHeaders(csrfToken: String?): Headers {
+        if (csrfToken == null) {
+            return xhrHeaders
+        }
+        return xhrHeaders.newBuilder()
+            .add("X-Csrf-Token", csrfToken)
+            .build()
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        // Sidebar mangas should always override any other search, so they should appear first
+        // in the case/when and only propagate to super when a "normal" search is issued
+        val sortOrderFilter = filters.filterIsInstance<SortOrderFilter>().firstOrNull()
+
+        return when {
+            sortOrderFilter?.state == 2 ->
+                sidebarRequest("top_rated")
```

Comment:
Could we put this into a map then get the URI part directly instead of checking each case? Also use that map to populate `getSortOrderURIs`